src/services/refactors/convertStringOrTemplateLiteral.ts

```diff
@@ -0,0 +1,190 @@
+    }
+
+    function isStringConcatenationValid(node: Node): boolean {
+        const { containsString, areOperatorsValid } = treeToArray(node);
+        return containsString && areOperatorsValid;
+    }
+
+    function transformTreeToArray(node: Node): ReadonlyArray<Expression> {
+        return treeToArray(node).nodes;
+    }
+
+    function treeToArray(node: Node): { nodes: ReadonlyArray<Expression>, containsString: boolean, areOperatorsValid: boolean} {
+        if (isBinaryExpression(node)) {
+            const { nodes: leftNodes, containsString: leftHasString, areOperatorsValid: leftOperatorValid } = treeToArray(node.left);
+            const { nodes: rightNodes, containsString: rightHasString, areOperatorsValid: rightOperatorValid } = treeToArray(node.right);
+
+            if (!leftHasString && !rightHasString) {
+                return { nodes: [node], containsString: false, areOperatorsValid: true };
+            }
+
+            const currentOperatorValid = node.operatorToken.kind === SyntaxKind.PlusToken;
+            const areOperatorsValid = leftOperatorValid && currentOperatorValid && rightOperatorValid;
+
+            return { nodes: leftNodes.concat(rightNodes), containsString: true, areOperatorsValid };
+        }
+
+        return { nodes: [node as Expression], containsString: isStringLiteral(node), areOperatorsValid: true };
+    }
+
+    function concatConsecutiveString(index: number, nodes: ReadonlyArray<Expression>): [number, string] {
+        let text = "";
+
+        while (index < nodes.length && isStringLiteral(nodes[index])) {
+            text = text + decodeRawString(nodes[index].getText());
+            index++;
+        }
+
+        text = escapeText(text);
+        return [index, text];
+    }
+
+    function nodesToTemplate(nodes: ReadonlyArray<Expression>) {
+        const templateSpans: TemplateSpan[] = [];
+        const [begin, headText] = concatConsecutiveString(0, nodes);
+        const templateHead = createTemplateHead(headText);
+
+        if (begin === nodes.length) return createNoSubstitutionTemplateLiteral(headText);
+
+        for (let i = begin; i < nodes.length; i++) {
+            const expression = isParenthesizedExpression(nodes[i]) ? (nodes[i] as ParenthesizedExpression).expression : nodes[i];
+            const [newIndex, subsequentText] = concatConsecutiveString(i + 1, nodes);
+            i = newIndex - 1;
+
+            const templatePart = i === nodes.length - 1 ? createTemplateTail(subsequentText) : createTemplateMiddle(subsequentText);
+            templateSpans.push(createTemplateSpan(expression, templatePart));
+        }
+
+        return createTemplateExpression(templateHead, templateSpans);
+    }
+
+    const octalToUnicode = (_match: string, grp: string) => String.fromCharCode(parseInt(grp, 8));
+
+    function decodeRawString(content: string) {
+        const outerQuotes = /["']((.|\s)*)["']/;
+        const octalEscape = /\\((?:[1-7][0-7]{0,2}|[0-7]{2,3}))/g;

```

Comment:
i think you would also have to deal with other escape sequences such as unicode here, but you might not need this function as per my comment above.